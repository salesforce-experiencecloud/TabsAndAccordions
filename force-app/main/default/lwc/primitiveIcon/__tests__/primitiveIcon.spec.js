/*
 * Copyright (c) 2019, salesforce.com, inc.
 * All rights reserved.
 * SPDX-License-Identifier: MIT
 * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
 */

import { createElement } from 'lwc';
import Element from 'c/primitiveIcon';

const createPrimitiveIcon = () => {
    const element = createElement('c-primitive-icon', { is: Element });
    document.body.appendChild(element);
    return element;
};

describe('c-primitive-icon', () => {
    it('default (size=medium, variant="", svg renders but is blank)', () => {
        const element = createPrimitiveIcon();
        expect(element).toMatchSnapshot();
    });

    describe('should render as expected for size', () => {
        it('xx-small', () => {
            const element = createPrimitiveIcon();
            element.iconName = 'utility:text_background_color';
            element.size = 'xx-small';
            return Promise.resolve().then(() => {
                expect(element).toMatchSnapshot();
            });
        });

        it('x-small', () => {
            const element = createPrimitiveIcon();
            element.iconName = 'utility:text_background_color';
            element.size = 'x-small';
            return Promise.resolve().then(() => {
                expect(element).toMatchSnapshot();
            });
        });

        it('small', () => {
            const element = createPrimitiveIcon();
            element.iconName = 'utility:text_background_color';
            element.size = 'small';
            return Promise.resolve().then(() => {
                expect(element).toMatchSnapshot();
            });
        });

        it('medium', () => {
            const element = createPrimitiveIcon();
            element.iconName = 'utility:text_background_color';
            element.size = 'medium';
            return Promise.resolve().then(() => {
                expect(element).toMatchSnapshot();
            });
        });

        it('large', () => {
            const element = createPrimitiveIcon();
            element.iconName = 'utility:text_background_color';
            element.size = 'large';
            return Promise.resolve().then(() => {
                expect(element).toMatchSnapshot();
            });
        });

        it('change from small to large', () => {
            const element = createPrimitiveIcon();
            element.iconName = 'utility:text_background_color';
            element.size = 'small';
            return Promise.resolve()
                .then(() => {
                    element.size = 'large';
                })
                .then(() => {
                    expect(element).toMatchSnapshot();
                });
        });
    });

    describe('should render as expected for variant', () => {
        it('bare', () => {
            const element = createPrimitiveIcon();
            element.iconName = 'utility:salesforce1';
            element.variant = 'bare';
            return Promise.resolve().then(() => {
                expect(element).toMatchSnapshot();
            });
        });

        it('error', () => {
            const element = createPrimitiveIcon();
            element.iconName = 'utility:salesforce1';
            element.variant = 'error';
            return Promise.resolve().then(() => {
                expect(element).toMatchSnapshot();
            });
        });

        it('inverse', () => {
            const element = createPrimitiveIcon();
            element.iconName = 'utility:salesforce1';
            element.variant = 'inverse';
            return Promise.resolve().then(() => {
                expect(element).toMatchSnapshot();
            });
        });

        it('warning', () => {
            const element = createPrimitiveIcon();
            element.iconName = 'utility:salesforce1';
            element.variant = 'warning';
            return Promise.resolve().then(() => {
                expect(element).toMatchSnapshot();
            });
        });

        it('change from inverse to error', () => {
            const element = createPrimitiveIcon();
            element.iconName = 'utility:salesforce1';
            element.variant = 'inverse';
            return Promise.resolve()
                .then(() => {
                    element.variant = 'error';
                })
                .then(() => {
                    expect(element).toMatchSnapshot();
                });
        });
    });

    it('iconName value change from utility:salesforce1 to standard:account', () => {
        const element = createPrimitiveIcon();
        element.iconName = 'utility:salesforce1';
        return Promise.resolve()
            .then(() => {
                element.iconName = 'standard:account';
            })
            .then(() => {
                expect(element).toMatchSnapshot();
            });
    });
});
