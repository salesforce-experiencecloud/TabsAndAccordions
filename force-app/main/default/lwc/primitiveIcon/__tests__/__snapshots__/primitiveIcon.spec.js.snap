// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`c-primitive-icon default (size=medium, variant="", svg renders but is blank) 1`] = `
<c-primitive-icon>
  #shadow-root(open)
    <svg
      aria-hidden="true"
      class="slds-icon slds-icon-text-default"
      data-key=""
      focusable="false"
    >
      <use
        xlink:href=""
      />
    </svg>
</c-primitive-icon>
`;

exports[`c-primitive-icon iconName value change from utility:salesforce1 to standard:account 1`] = `
<c-primitive-icon>
  #shadow-root(open)
    <svg
      aria-hidden="true"
      class="slds-icon slds-icon-text-default"
      data-key="account"
      focusable="false"
    >
      <use
        xlink:href="/assets/icons/standard-sprite/svg/symbols.svg#account"
      />
    </svg>
</c-primitive-icon>
`;

exports[`c-primitive-icon should render as expected for size change from small to large 1`] = `
<c-primitive-icon>
  #shadow-root(open)
    <svg
      aria-hidden="true"
      class="slds-icon slds-icon-text-default slds-icon_large"
      data-key="text_background_color"
      focusable="false"
    >
      <use
        xlink:href="/assets/icons/utility-sprite/svg/symbols.svg#text_background_color"
      />
    </svg>
</c-primitive-icon>
`;

exports[`c-primitive-icon should render as expected for size large 1`] = `
<c-primitive-icon>
  #shadow-root(open)
    <svg
      aria-hidden="true"
      class="slds-icon slds-icon-text-default slds-icon_large"
      data-key="text_background_color"
      focusable="false"
    >
      <use
        xlink:href="/assets/icons/utility-sprite/svg/symbols.svg#text_background_color"
      />
    </svg>
</c-primitive-icon>
`;

exports[`c-primitive-icon should render as expected for size medium 1`] = `
<c-primitive-icon>
  #shadow-root(open)
    <svg
      aria-hidden="true"
      class="slds-icon slds-icon-text-default"
      data-key="text_background_color"
      focusable="false"
    >
      <use
        xlink:href="/assets/icons/utility-sprite/svg/symbols.svg#text_background_color"
      />
    </svg>
</c-primitive-icon>
`;

exports[`c-primitive-icon should render as expected for size small 1`] = `
<c-primitive-icon>
  #shadow-root(open)
    <svg
      aria-hidden="true"
      class="slds-icon slds-icon-text-default slds-icon_small"
      data-key="text_background_color"
      focusable="false"
    >
      <use
        xlink:href="/assets/icons/utility-sprite/svg/symbols.svg#text_background_color"
      />
    </svg>
</c-primitive-icon>
`;

exports[`c-primitive-icon should render as expected for size x-small 1`] = `
<c-primitive-icon>
  #shadow-root(open)
    <svg
      aria-hidden="true"
      class="slds-icon slds-icon-text-default slds-icon_x-small"
      data-key="text_background_color"
      focusable="false"
    >
      <use
        xlink:href="/assets/icons/utility-sprite/svg/symbols.svg#text_background_color"
      />
    </svg>
</c-primitive-icon>
`;

exports[`c-primitive-icon should render as expected for size xx-small 1`] = `
<c-primitive-icon>
  #shadow-root(open)
    <svg
      aria-hidden="true"
      class="slds-icon slds-icon-text-default slds-icon_xx-small"
      data-key="text_background_color"
      focusable="false"
    >
      <use
        xlink:href="/assets/icons/utility-sprite/svg/symbols.svg#text_background_color"
      />
    </svg>
</c-primitive-icon>
`;

exports[`c-primitive-icon should render as expected for variant bare 1`] = `
<c-primitive-icon>
  #shadow-root(open)
    <svg
      aria-hidden="true"
      class=""
      data-key="salesforce1"
      focusable="false"
    >
      <use
        xlink:href="/assets/icons/utility-sprite/svg/symbols.svg#salesforce1"
      />
    </svg>
</c-primitive-icon>
`;

exports[`c-primitive-icon should render as expected for variant change from inverse to error 1`] = `
<c-primitive-icon>
  #shadow-root(open)
    <svg
      aria-hidden="true"
      class="slds-icon slds-icon-text-error"
      data-key="salesforce1"
      focusable="false"
    >
      <use
        xlink:href="/assets/icons/utility-sprite/svg/symbols.svg#salesforce1"
      />
    </svg>
</c-primitive-icon>
`;

exports[`c-primitive-icon should render as expected for variant error 1`] = `
<c-primitive-icon>
  #shadow-root(open)
    <svg
      aria-hidden="true"
      class="slds-icon slds-icon-text-error"
      data-key="salesforce1"
      focusable="false"
    >
      <use
        xlink:href="/assets/icons/utility-sprite/svg/symbols.svg#salesforce1"
      />
    </svg>
</c-primitive-icon>
`;

exports[`c-primitive-icon should render as expected for variant inverse 1`] = `
<c-primitive-icon>
  #shadow-root(open)
    <svg
      aria-hidden="true"
      class="slds-icon"
      data-key="salesforce1"
      focusable="false"
    >
      <use
        xlink:href="/assets/icons/utility-sprite/svg/symbols.svg#salesforce1"
      />
    </svg>
</c-primitive-icon>
`;

exports[`c-primitive-icon should render as expected for variant warning 1`] = `
<c-primitive-icon>
  #shadow-root(open)
    <svg
      aria-hidden="true"
      class="slds-icon slds-icon-text-warning"
      data-key="salesforce1"
      focusable="false"
    >
      <use
        xlink:href="/assets/icons/utility-sprite/svg/symbols.svg#salesforce1"
      />
    </svg>
</c-primitive-icon>
`;
